wordle-web/tests/wordService.test.ts

```diff
@@ -0,0 +1,63 @@
+import { expect, test } from "vitest";
+import { Game } from "~/scripts/game";
+import {WordList } from "~/scripts/wordList";
+import { LetterState } from "~/scripts/letter";
+import { WordService } from "~/scripts/wordService";
+
+test('WordService test full word', ()=> {
+  const guess = ['aargh'];
+  const states = [[
+    LetterState.Correct,
+    LetterState.Correct,
+    LetterState.Correct,
+    LetterState.Correct,
+    LetterState.Correct,
+  ]];
+
+  const validWords = WordService.validGuessedWords(guess, states);
+  expect(validWords).toEqual(['aargh']);
+});
+
+test('WordService test partial word', ()=> {
+  const guess = ['aarg'];
+  const states = [[
+    LetterState.Correct,
+    LetterState.Correct,
+    LetterState.Correct,
+    LetterState.Correct,
+    LetterState.Unknown,
+  ]];
```

Comment:
I think using a full word but with the last letter unknown would be a better test since we will never be able to submit a guess that is less than five letters based on `game` class.
```suggestion
test('WordService test unknown letter at end', ()=> {
  const guess = ['aargz'];
  const states = [[
    LetterState.Correct,
    LetterState.Correct,
    LetterState.Correct,
    LetterState.Correct,
    LetterState.Unknown,
  ]];
```

Reply:
refactored this part